src/BuildIntegration/Microsoft.NETCore.Native.Publish.targets

```diff
@@ -0,0 +1,59 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <BuildTasksPath Condition="'$(BuildTasksPath)' == ''">$(MSBuildThisFileDirectory)..\tools\ILCompiler.Build.Tasks.dll</BuildTasksPath>
+    
+    <!--
+      Prevent dotnet CLI from deploying the CoreCLR shim executable since we produce
+      a native executable.
+    -->
+    <DeployAppHost>false</DeployAppHost>
+  </PropertyGroup>
+
+  <Target Name="_ComputeIlcCompileInputs"
+          BeforeTargets="ComputeIlcCompileInputs">
+    <ItemGroup>
+      <IlcReference Include="@(_ManagedResolvedAssembliesToPublish)" />
+    </ItemGroup>
+  </Target>
+
+  <!--
+    This target hooks into the dotnet CLI publish pipeline. That pipeline has
+    a target called ComputeFilesToPublish which produces the ItemGroup
+    ResolvedFileToPublish based on the inputs of @(IntermediateAssembly)
+    and @(ResolvedAssembliesToPublish). We modify those two item groups
+    to control what gets published after CoreRT optimizes the application.
+  -->
+  <Target Name="ComputeLinkedFilesToPublish"
+          BeforeTargets="ComputeFilesToPublish"
+          DependsOnTargets="_ComputeAssembliesToCompileToNative;LinkNative">
+    
+    <ItemGroup>
+      <ResolvedAssembliesToPublish Remove="@(_AssembliesToSkipPublish)" />
+      <ResolvedAssembliesToPublish Include="@(_LinkedResolvedAssemblies)" />
+    </ItemGroup>
+    
+    <ItemGroup>
+      <_NativeIntermediateAssembly Include="@(IntermediateAssembly->'$(NativeOutputPath)%(Filename).exe')" />
```

Comment:
I think msbuild has a way to express the `.exe` that is hardcoded here. This won't work for #4718.